js/adapt-stateful-session.js

```diff
@@ -127,6 +145,16 @@ define([
 			Adapt.offlineStorage.set("interaction", id, response, result, latency, responseType);
 		},
 
+		onResetSession: function () {
+			this.removeEventListeners();
+			this.setupEventListeners();
+			var sessionPairs = this.getSessionState();
+			
+			// hard status reset
+			Adapt.offlineStorage.set("status", "incomplete");
```

Comment:
I'm not 100% sure we should reset the lesson status here. At the very least I think it should probably be a config setting.
In fact, it's arguable that all this should be configurable - after all, if you did have a course where all languages have the same structure then technically there's no reason to reset.